tests/phpunit/tests/block-supports/wpRenderElementsSupport.php

```diff
@@ -18,6 +18,134 @@ public function tear_down() {
 		parent::tear_down();
 	}
 
+	/**
+	 * Tests that no output is generated when a block's markup is empty.
+	 *
+	 * @ticket 59578
+	 *
+	 * @dataProvider data_elements_block_support_class
+	 *
+	 * @covers ::wp_render_elements_support
+	 *
+	 * @param array  $color_settings  The color block support settings used for elements support.
+	 * @param array  $elements_styles The elements styles within the block attributes.
+	 * @param string $block_markup    Original block markup.
+	 * @param string $expected_markup Resulting markup after application of elements block support.
+	 *
+	 * @return void
+	 */
+	public function test_leaves_empty_block_content_empty( $color_settings, $elements_styles, $block_markup, $expected_markup ) {
```

Comment:
The `@dataProvider` annotation will cause all the data sets in `data_elements_block_support_class` to be run through this test which I don't think is what was intended.

Additionally, we could add an extra entry to the data provider to cover the empty content scenario removing this test function entirely.

Reply:
> will cause all the data sets in data_elements_block_support_class to be run through this test which I don't think is what was intended.

it was what I intended, but I will strip it away to move this forward :) I only meant to test that in the normal selection of existing cases we're testing, that the empty content/empty attributes/etc… all behave in the same predictable way for those cases.